third-party/realdds/src/dds-device-server.cpp

```diff
@@ -237,114 +256,140 @@ void dds_device_server::on_control_message_received()
+                    realdds::print( sample.sample_identity.writer_guid(), false ),  // raw guid
+                    sample.sample_identity.sequence_number().to64long(),
+                } );
                 try
                 {
-                    handle_control_message( x );
+                    std::string id = rsutils::json::get< std::string >( j, id_key );
+                    reply[id_key] = id;
+                    reply[control_key] = j;
+                    handle_control_message( id, j, reply );
                 }
                 catch( std::exception const & e )
                 {
-                    publish_notification( json{
-                        { "id", "error" },
-                        { "error", e.what() },
-                        { "original", x.json_data() },
-                    } );
+                    reply[status_key] = "error";
+                    reply[explanation_key] = e.what();
+                }
+                try
+                {
+                    LOG_DEBUG( "----->   reply " << reply );
+                    publish_notification( reply );
+                }
+                catch( ... )
+                {
+                    LOG_ERROR( "failed to send reply" );
                 }
             } );
     }
 }
 
-void dds_device_server::handle_control_message( topics::flexible_msg control_message )
+
+void dds_device_server::handle_control_message( std::string const & id,
+                                                nlohmann::json const & j,
+                                                nlohmann::json & reply )
 {
-    auto j = control_message.json_data();
-    auto id = rsutils::json::get< std::string >( j, "id" );
-    LOG_DEBUG( "-----> JSON = " << j.dump() );
+    LOG_DEBUG( "<----- control " << j );
 
-    if( id.compare( "open-streams" ) == 0 )
+    if( id.compare( id_open_streams ) == 0 )
     {
         if ( _open_streams_callback )
             _open_streams_callback( j );
     }
-    else if( id.compare( "set-option" ) == 0 )
+    else if( id.compare( id_set_option ) == 0 )
     {
-        handle_set_option( j );
+        handle_set_option( j, reply );
     }
-    else if( id.compare( "query-option" ) == 0 )
+    else if( id.compare( id_query_option ) == 0 )
     {
-        handle_query_option( j );
+        handle_query_option( j, reply );
     }
     else
     {
         DDS_THROW( runtime_error, "invalid control '" + id + "'" );
     }
 }
 
-void dds_device_server::handle_set_option( const nlohmann::json & j )
+
+void dds_device_server::handle_set_option( const nlohmann::json & j, nlohmann::json & reply )
 {
-    const std::string & option_name = rsutils::json::get< std::string >( j, "option-name" );
-    const std::string & owner_name = rsutils::json::get< std::string >( j, "owner-name" );
-    uint32_t counter = rsutils::json::get< uint32_t >( j, "counter" );
+    auto option_name = rsutils::json::get< std::string >( j, option_name_key );
```

Comment:
Was of type `const std::string &`, why did you remove the `&`? Isn't it more efficient? Aren't we allowed to reference the json?

Reply:
`rsutils::json::get<>()` does not return a reference... right?

Reply:
You are right. A reference type will reference a temporary returned object (extending the temporary life time) but a non reference type can result in a "return value optimization" moving the returned temporary, or even better using copy elision (mandatory from C++17)